daemon/daemonbuilder/builder.go

```diff
@@ -29,12 +28,29 @@ import (
 
 // Docker implements builder.Docker for the docker Daemon object.
 type Docker struct {
-	Daemon      *daemon.Daemon
+	Daemon      Builder
 	OutOld      io.Writer
 	AuthConfigs map[string]cliconfig.AuthConfig
 	Archiver    *archive.Archiver
 }
 
+// Builder defines an interface that the responsible for building images must implement
+type Builder interface {
```

Comment:
@tiborvass what's your suggestions to handle this interface? It's not really the Docker interface, but the interface the daemon implements. Should we unify them?